Cannot use v1 model as type parameter in v2 generic model

### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

When trying to parametrise a generic pydantic v2 model using a v1 model, pydantic fails with a cryptic error message that does not indicate what the problem is:

```
tests/unit/test_.py:19 (test_v1)
def test_v1():
>       assert Container[V1](data=V1())
E       TypeError: validate() takes 2 positional arguments but 3 were given

test_.py:21: TypeError
```

I see a couple of resolution paths:
- indicate in the migration docs that this is not supported
- fail with a meaningful error message
- support v1 models as parametrisation types

### Example Code

```Python
import typing

import pydantic.v1

DataT = typing.TypeVar("DataT")


class V1(pydantic.v1.BaseModel):
    val: list[int] = [1, 2, 3]


class V2(pydantic.BaseModel):
    val: list[int] = [1, 2, 3]


class Container(pydantic.BaseModel, typing.Generic[DataT]):
    data: DataT


def test_v1():
    assert Container[V1](data=V1())


def test_v2():
    assert Container[V2](data=V2())
```


### Python, Pydantic & OS Version

```Text
pydantic version: 2.4.0
        pydantic-core version: 2.10.0
          pydantic-core build: profile=release pgo=false
                 install path: /Users/xxx/Library/Caches/pypoetry/virtualenvs/juiceutils-base-sT27uenX-py3.9/lib/python3.9/site-packages/pydantic
               python version: 3.9.15 (main, Feb 20 2023, 14:56:22)  [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-13.4.1-arm64-arm-64bit
             related packages: fastapi-0.103.1 email-validator-2.0.0.post2 pydantic-extra-types-2.1.0 typing_extensions-4.7.1 pydantic-settings-2.0.3
```

--- a/docs/migration.md
+++ b/docs/migration.md
@@ -134,6 +134,9 @@ The `pydantic.generics.GenericModel` class is no longer necessary, and has been 
 create generic `BaseModel` subclasses by just adding `Generic` as a parent class on a `BaseModel` subclass directly.
 This looks like `class MyGenericModel(BaseModel, Generic[T]): ...`.
 
+Mixing of V1 and V2 models is not supported which means that type parameters of such generic `BaseModel` (V2)
+cannot be V1 models.
+
 While it may not raise an error, we strongly advise against using _parametrized_ generics in `isinstance` checks.
 
   * For example, you should not do `isinstance(my_model, MyGenericModel[int])`.

